src/structures/ActionRow.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+const BaseComponent = require('./BaseComponent');
+const Components = require('../ComponentFactory');
+const { ComponentTypes } = require('../util/Constants');
+
+/**
+ * Represents an action row containing components.
+ * @extends {BaseComponent}
+ */
+class ActionRow extends BaseComponent {
+  /**
+   * Components that can be placed in an action row
+   * * Button
+   * * SelectMenu
+   * * InputText
+   * @typedef {Button|SelectMenu|InputText} ActionRowComponent
+   */
+
+  /**
+   * Options for components that can be placed in an action row
+   * * ButtonOptions
+   * * SelectMenuOptions
+   * * InputTextOptions
+   * @typedef {ButtonOptions|SelectMenuOptions|InputTextOptions} ActionRowComponentOptions
+   */
+
+  /**
+   * Data that can be resolved into components that can be placed in an action row
+   * * ActionRowComponent
+   * * ActionRowComponentOptions
+   * @typedef {ActionRowComponent|ActionRowComponentOptions} ActionRowComponentResolvable
+   */
+
+  /**
+   * @typedef {BaseComponentOptions} ActionRowOptions
+   * @property {ActionRowComponentResolvable[]} [components]
+   * The components to place in this action row
+   */
```

Comment:
Please put these in places where they’re relevant so it’s easier to find and edit them later on (please do the same for all files)

Reply:
How is it not relevant here in this file?

Reply:
To be honest this file is good but some of the others have all of the typedefs at the very top where they could be scattered through the file as they are used

Reply:
This is the same format being used beforehand

Reply:
Alright then